### ISSUE TITLE
Malformed log group name when used with terraform-aws-lambda-function

### ISSUE THREAD

**Issue by MReggler** [cloudposse/terraform-aws-cloudwatch-logs#52]
### Describe the Bug

When using [terraform-aws-lambda-function](https://github.com/cloudposse/terraform-aws-lambda-function) to create a lambda and associated log group, the resulting log groups can in some cases not match the function name, which results in no logs being sent to the created group.

### Expected Behavior

I create a lambda using the [terraform-aws-lambda-function](https://github.com/cloudposse/terraform-aws-lambda-function) module, and the name of this lambda contains an underscore.

```hcl
module "lambda" {
  source  = "cloudposse/lambda-function/aws"
  version = "0.5.3"

  function_name      = "my_function_name"
  ...

  context = module.this.context
}
```

The resultant log group created by this module should be called `/aws/lambda/my_function_name`

Instead the log group created is called  `/aws/lambda/myfunctionname`, as the label for the log group contains a `regex_replace_chars` rule that does not allow for underscores

https://github.com/cloudposse/terraform-aws-cloudwatch-logs/blob/f622326cce042d0e49b2613cc994ab710355ac7f/main.tf#L5C1-L13C2

### Steps to Reproduce

Invoke the lambda module with `var.function_name` set to a value that includes an underscore.

### Screenshots

![Screenshot 2024-02-20 at 16 32 36](https://github.com/cloudposse/terraform-aws-cloudwatch-logs/assets/25884556/1c5edd33-8e3a-4930-98b7-1b4257b22797)

### Environment

v0.6.6  -- version of the module used in the lambda module -- IS AFFECTED
v0.6.8  -- current version of the module -- IS AFFECTED

### Additional Context

This fix requires a bump to the version of the cloudwatch logs module used in the lambda module (and/or anywhere else in the CloudPosse module/component libraries that support the creation of resources with underscores in their names.

**Comment by nnsense** [comment]
I'm not sure it is related, but I'm also using lambda module and I also have end up with a CloudWatch log group with a different name from what is set by the lamdba module.
 
If I use this module alone I can't set the log group name to something in PascalCase, something somewhere is setting the name to `lower`, as if `label_value_case` was set to `lower` by default.. actually, it is stated into `context.tf` that the default is `lower`, but then `default` is actually set to `null` in there, so I really don't know from where that `lower` is coming from.

Anyway if I set the name like this:
```
name              = "infra-alerts-CamelCase"
```

I get this:
```
[...]
  + resource "aws_cloudwatch_log_group" "default" {
  + arn               = (known after apply)                                                                                                                                                                      
  + id                = (known after apply)
  + log_group_class   = (known after apply)                                                                                                                                                                      
  + name              = "/aws/lambda/infra-mgmt-ssolastlogintracking" <--------- lowercase
```

Like I said if I want that name to be CamelCase I need to force `label_value_case = "null"`, and I get what I want but, if I use lambda, I cannot set `label_value_case` (I tried and it's not passed to the submobdule apparently) so I eventually get the cloudwatch name different from the function name, which leads exactly to the same issue as above.

**Comment by nnsense** [comment]
Right, investigated a bit and I really think these issues are due to how the lamdba is setting this module, unrelated this specific module. That is where the regex is removing the underscore and it is not dealing with specific "case" settings. I'm creating an issue there and will create a patch for these use cases.

**Comment by globart** [comment]
@MReggler this should be solved in [v0.6.9](https://github.com/cloudposse/terraform-aws-cloudwatch-logs/releases/tag/v0.6.9) (check Bug Fixes section)
